File: abstractions/php/src/Authentication/AccessTokenProvider.php
<?php
/**
 * Copyright (c) Microsoft Corporation.  All Rights Reserved.
 * Licensed under the MIT License.  See License in the project root
 * for license information.
 */


namespace Microsoft\Kiota\Abstractions\Authentication;

use Http\Promise\Promise;

/**
 * Interface AccessTokenProvider
 *
 * Defines a contract for obtaining access tokens for a given url
 *
 * @package Microsoft\Kiota\Abstractions\Authentication
 * @copyright 2022 Microsoft Corporation
 * @license https://opensource.org/licenses/MIT MIT License
 * @link https://developer.microsoft.com/graph
 */
interface AccessTokenProvider
{
    /**
     * Method called by {@link BaseBearerTokenAuthenticationProvider} to get the access token
     * Returns a Promise that is fulfilled with the access token string
     *
     * @param string $url the target URI to get an access token for
     * @return Promise
     */
    public function getAuthorizationTokenAsync(string $url): Promise;
}

File: abstractions/python/kiota/abstractions/authentication/access_token_provider.py
from abc import ABC, abstractmethod

from .allowed_hosts_validator import AllowedHostsValidator


class AccessTokenProvider(ABC):
    """Defines a contract for obtaining access tokens for a given url.
    """

    @abstractmethod
    async def get_authorization_token(self, uri: str) -> str:
        """This method is called by the BaseBearerTokenAuthenticationProvider class to get the
        access token.

        Args:
            uri (str): The target URI to get an access token for.
        Returns:
            str: The access token to use for the request.
        """
        pass

    @abstractmethod
    def get_allowed_hosts_validator(self) -> AllowedHostsValidator:
        """Retrieves the allowed hosts validator.

        Returns:
            AllowedHostsValidator: The allowed hosts validator.
        """
        pass
